app/models/partners/profile.rb

```diff
@@ -114,7 +114,8 @@ class Profile < Base
     ]
 
     def client_share_total
-      served_areas.sum(&:client_share)
+      # client_share could be nil
+      served_areas.sum { |area| area.client_share.to_i }
```

Comment:
Might be a bit more natural to do `served_areas.compact.sum(&:client_share)`.

Reply:
Good idea, I forgot compact was a thing

Reply:
let me squash my commits too, one sec

Reply:
Alright pushed the change

Reply:
Hmm.. so when I tried with compact the error came back 
<img width="1792" alt="image" src="https://github.com/rubyforgood/human-essentials/assets/51760107/162da140-7fe2-4e0b-9b1d-81cc9af800a8">

```
10:08:54 web.1    |   TRANSACTION (0.5ms)  ROLLBACK
10:08:54 web.1    |   ↳ app/services/partner_profile_update_service.rb:27:in `perform_profile_service'
10:08:54 web.1    | [!] PartnerProfileUpdateService failed to update profile for 1: ["Served areas client share is not a number", "Served areas client share is not included in the list"] [#<TypeError: nil can't be coerced into Integer>]
```

Reply:
Let me try adding a test on the profile.rb model level, that might be more likely to catch the error. because the partner_profile_update_service function is checking the stringified errors from the result

Reply:
I don't really see how that's possible... `compact` should be deleting all `nil` values, so you shouldn't be able to get a TypeError :open\_mouth: